### DISCUSSION TITLE
`hostRules.timeout` and `external-host-error`

### DISCUSSION THREAD

**Discussion by Leksat** [renovatebot/renovate#20186] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

Renovate works.

#### Initial issue

About one-third of Renovate jobs were failing with `external-host-error`: https://app.renovatebot.com/dashboard#github/AmazeeLabs/silverback-mono (from oldest to 997332772 included)

Each time it was the same NPM package - `@docusaurus/core` ([logs](https://app.renovatebot.com/dashboard#github/AmazeeLabs/silverback-mono/996731394)). The URL it was trying to get - https://registry.npmjs.org/@docusaurus%2Fcore - is a 8mb json file. No issues with the URL locally.

#### Attempt 1

I banned a part of monorepo that had this dependency:
```json
{ "ignorePaths": ["docs/**"] }
```

Then I got `external-host-error` for a different package - `@graphql-tools/load` ([logs](https://app.renovatebot.com/dashboard#github/AmazeeLabs/silverback-mono/997358681)). URL: https://registry.npmjs.org/@graphql-tools%2Fload (3.6mb)

#### Attempt 2

Then I tried to increase the timeout:
```json
{ "hostRules": [ { "timeout": 120000 } ] }
```

I know it is not recommended, but I just wanted to see if it helps.

#### Result

The next jobs passed.

#### Question

Is it fine to keep the increased `timeout`?

**Comment by rarkins** [comment, marked as answer]
Certain packages, which usually seem to be the large ones but not always the largest (for example many popular ones are >100MB). It seems to be triggered by certain repos with too many simultaneous requests at a time, and we don't know why (e.g. a network gateway drops connections or something).

Yes, it's fine to keep an increased timeout. Also, Renovate should be caching successful results for 24 hours so in case a future fetch fails it should fall back to the cached result after the timeout/error.
